scripts/NFTVesting/test-deploy/interval/deploy_interval_time_lock.js

```diff
@@ -0,0 +1,169 @@
+	const ownerBalanceAfter = await ethers.provider.getBalance(nftLocker.address);
+	console.log(
+		"nftLocker Balance After: ",
+		ethers.utils.formatEther(ownerBalanceAfter)
+	);
+
+	// check NFT Locked
+	const nftLocked = await timeLockInstance.nft();
+	console.log("nftLocked: ", nftLocked);
+
+	// Send NFT to timelock contract
+	await basicNFTInstance.transferFrom(
+		nftLocker.address,
+		timeLockInstance.address,
+		0 // token id
+	);
+
+	// check nftLocker of NFT after transfer to be timelock contract
+	const nftOwnerAfterTransfer = await basicNFTInstance.ownerOf(0);
+	console.log("nftOwnerAfterTransfer: ", nftOwnerAfterTransfer);
+
+	// Set new timestamp by speeding up time
+	await ethers.provider.send("evm_setNextBlockTimestamp", [
+		timestampBefore + 1111,
+	]);
+	await ethers.provider.send("evm_mine"); // Fast forward time
+
+	// Get new timestamp
+	const blockNumAfter = await ethers.provider.getBlockNumber();
+	const blockAfter = await ethers.provider.getBlock(blockNumAfter);
+	const currentTimeStamp = blockAfter.timestamp;
+	console.log("currentTimeStamp: ", currentTimeStamp, "\n");
+
+	// Get balance of timelock contract
+	const timeLockBalance = await ethers.provider.getBalance(
+		timeLockInstance.address
+	);
+	console.log("timeLockBalance: ", ethers.utils.formatEther(timeLockBalance));
+
+	// Get current discount
+	const currentDiscount = await timeLockInstance.getDiscount();
+	console.log(
+		"currentDiscount: ",
+		ethers.utils.formatEther(currentDiscount),
+		"\n"
+	);
+
+	// release NFT
+	console.log(
+		"Release Time: ",
+		ethers.utils.formatUnits(await timeLockInstance.vestingStartTime(), 18 - 18)
+	);
+
+	console.log(
+		"Release Time: ",
+		ethers.utils.formatUnits(await timeLockInstance.vestingStartTime(), 18 - 18)
+	);
+
+	const blockNumBeforeRelease = await ethers.provider.getBlockNumber();
+	const blockBeforeRelease = await ethers.provider.getBlock(
+		blockNumBeforeRelease
+	);
+	console.log("Time Before Release:", blockBeforeRelease.timestamp, "\n");
+
+	// Get current discount
+	const currentDiscount2 = await timeLockInstance.getDiscount();
+	console.log(
+		"currentDiscount: ",
+		ethers.utils.formatEther(currentDiscount2),
+		"\n"
+	);
+
+	console.log("Releasing NFT... \n");
+
+	const releaseTx = await timeLockInstance.release(); // failing
+	const newNftOwner = await basicNFTInstance.ownerOf(0);
+	console.log("newNftOwner: ", newNftOwner); // same as original beneficiary
+
+	// Get the ETH balance of NFT Locker and Beneficiary
+	const nftLockerBalanceAfter = await ethers.provider.getBalance(
+		nftLocker.address
+	);
+	const beneficiaryBalanceAfter = await ethers.provider.getBalance(
+		beneficiary.address
+	);
+
+	console.log(
+		"nftLockerBalanceAfter: ",
+		ethers.utils.formatEther(nftLockerBalanceAfter)
+	);
+	console.log(
+		"beneficiaryBalanceAfter: ",
+		ethers.utils.formatEther(beneficiaryBalanceAfter)
+	);
+
+	// // check if {release} function can be called again. Expected failure.
```

Comment:
Don't need right? Or already tested in another script?

Reply:
actually i dont think its needed